server/hl7/hl7.controller.js

```diff
@@ -104,11 +104,37 @@ function getUserFiles(req, res, next) {
   if (req.user.files.length === 0) {
     err = new APIError('User has no files uploaded', httpStatus.NO_CONTENT);
   } else {
-    err = new APIError('There was an error retrieving uploaded files', httpStatus.BAD_REQUEST);
+    err = new APIError(
+      'There was an error retrieving uploaded files',
+      httpStatus.BAD_REQUEST
+    );
   }
   return next(err);
 }
 
+function getParsedMessages(req, res, next) {
+  if (!mongoose.Types.ObjectId.isValid(req.params.fileId)) {
+    const error = new APIError('Error: Not a valid file ID', httpStatus.BAD_REQUEST);
+    next(error);
+    return;
+  }
+  Message.find({ fileId: req.params.fileId }, (err, docs) => {
+    if (docs.length === 0) {
+      const error = new APIError(
+        'Error: File Not Found!',
+        httpStatus.NOT_FOUND
+      );
+      return next(error);
+    }
+    const parsedMessages = docs.map(message => ({
```

Comment:
I think we actually want to return the entire message object in this case. We can probably just send `docs` as is instead of pruning in here

Reply:
Hmmm, yeah rereading the ticket and thinking about what this might be used for, I'm having second thoughts. This is good.